zookeeper-server/src/test/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProviderTest.java

```diff
@@ -0,0 +1,173 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.server.auth.znode.groupacl;
+
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.PortAssignment;
+import org.apache.zookeeper.ZKTestCase;
+import org.apache.zookeeper.ZKUtil;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.data.Id;
+import org.apache.zookeeper.server.MockServerCnxn;
+import org.apache.zookeeper.server.ServerCnxnFactory;
+import org.apache.zookeeper.server.ZooKeeperServer;
+import org.apache.zookeeper.server.auth.ServerAuthenticationProvider;
+import org.apache.zookeeper.server.auth.X509AuthenticationUtil;
+import org.apache.zookeeper.test.ClientBase;
+import org.apache.zookeeper.test.X509AuthTest;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class X509ZNodeGroupAclProviderTest extends ZKTestCase {
+  private static final Logger LOG = LoggerFactory.getLogger(X509ZNodeGroupAclProviderTest.class);
+  private static final String HOSTPORT = "127.0.0.1:" + PortAssignment.unique();
+  private static X509AuthTest.TestCertificate domainXCert;
+  private static X509AuthTest.TestCertificate superCert;
+  private static X509AuthTest.TestCertificate unknownCert;
+  private static final String CLIENT_CERT_ID_SAN_MATCH_TYPE = "6";
+  private static final String SCHEME = "x509";
+  private static ZooKeeperServer zks;
+  private ServerCnxnFactory serverCnxnFactory;
+  private ZooKeeper admin;
+  private static final String AUTH_PROVIDER_PROPERTY_NAME = "zookeeper.authProvider.x509";
+  private static final String CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH = "/_CLIENT_URI_DOMAIN_MAPPING";
+  private static final String[] MAPPING_PATHS = {CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH,
+      CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH + "/CrossDomain",
+      CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH + "/CrossDomain/SuperUser",
+      CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH + "/DomainX",
+      CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH + "/DomainX/DomainXUser",
+      CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH + "/DomainY",
+      CLIENT_URI_DOMAIN_MAPPING_ROOT_PATH + "/DomainY/DomainYUser"};
+  private static final Map<String, String> SYSTEM_PROPERTIES = new HashMap<>();
+    static {
```

Comment:
Nit: This works, but you can also use something like `ImmutableMap`. No need to change.